File: updatesproducer/tests/mock_updates_provider.py
from datetime import datetime

from updatesproducer.kafka.iupdates_provider import IUpdatesProvider
from updatesproducer.updateapi.update import Update


class MockUpdatesProvider(IUpdatesProvider):
    def get_updates(self, user_id: str):
        return [
            Update(
                content='Mock updateapi',
                author_id='Mock author',
                creation_date=datetime.now(),
                url='mockurl://updateapi.com',
                media=[],
                repost=False)
        ]

File: UpdatesProducer.Tests/MockUpdatesProvider.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Common;

namespace UpdatesProducer.Tests
{
    internal class MockUpdatesProvider : IUpdatesProvider
    {
        public Task<IEnumerable<Update>> GetUpdatesAsync(string userId)
        {
            IEnumerable<Update> updates = new Update[]
            {
                new()
                {
                    Content = $"mock update by {userId}",
                    AuthorId = userId,
                    CreationDate = DateTime.Now
                }
            };
            
            return Task.FromResult(updates);
        }
    }
}